microprofile/reactive-streams-operators-smallrye/cdi-provider/src/main/java/org/wildfly/extension/microprofile/reactive/streams/operators/cdi/ReactiveEngineProvider.java

```diff
@@ -43,7 +44,6 @@
      * @throws IllegalStateException if no implementations are found.
      */
     @Produces
-    @ApplicationScoped
```

Comment:
Keep this `@ApplicationScoped`. Removing the annotation means it will be `@Dependent` - the annotation is **not inherited** from the class itself. Using dependent would leads to creation of new bean of this type for every injection point whereas before you had one shared instance.

Reply:
Done -thanks for the feedback!